src/main/kotlin/no/nav/faktureringskomponenten/service/beregning/AntallBeregner.kt

```diff
@@ -0,0 +1,50 @@
+package no.nav.faktureringskomponenten.service.beregning
+
+import mu.KotlinLogging
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+private val log = KotlinLogging.logger { }
+
+class AntallBeregner(private val fom: LocalDate, private val tom: LocalDate) {
+
+    private val førsteMånedDager = fom.lengthOfMonth().toBigDecimal()
+    private val sisteMånedDager = tom.lengthOfMonth().toBigDecimal()
+    private val erSammeMånedOgÅr = fom.year == tom.year && fom.monthValue == tom.monthValue
+
+    fun beregn(): BigDecimal {
+        val total = beregnFørsteMånedProsent() + beregnMånederMellomProsent() + beregnSisteMånedProsent()
+        log.info("AntallBeregner fom: {}, tom: {} som gir angitt antall total: {}", fom, tom, total)
+        return total
+    }
+
+    private fun beregnFørsteMånedProsent(): BigDecimal {
+        return if (erSammeMånedOgÅr) {
+            (tom.dayOfMonth.toBigDecimal() - fom.dayOfMonth.toBigDecimal() + BigDecimal.ONE)
+                .divide(førsteMånedDager, 2, RoundingMode.HALF_UP)
+        } else {
+            (førsteMånedDager - fom.dayOfMonth.toBigDecimal() + BigDecimal.ONE)
+                .divide(førsteMånedDager, 2, RoundingMode.HALF_UP)
+        }
+    }
+
+    private fun beregnSisteMånedProsent(): BigDecimal {
+        return if (erSammeMånedOgÅr) {
```

Comment:
Tror ikke du trenger sjekke om det er samme måned og år for beregning av siste måneden.

Reply:
Jeg innser det, men valgte å ikke dele det opp ettersom det ble lettere å forstå koden.

```
    fun beregn(): BigDecimal {
        val total = beregnFørsteMånedProsent() + beregnMånederMellomProsent() + beregnSisteMånedProsent()
        log.info("AntallBeregner fom: {}, tom: {} som gir angitt antall total: {}", fom, tom, total)
        return total
    }
``` 

Hvis jeg tar det ut som at det er topp-nivå som avgjør det, istedet for hver enkel metode blir det å se slik ut:

```
    fun beregn(): BigDecimal {
        var total = beregnFørsteMånedProsent()
        if(!erSammeMånedOgÅr) { 
            total += beregnMånederMellomProsent() + beregnSisteMånedProsent()
        }
        log.info("AntallBeregner fom: {}, tom: {} som gir angitt antall total: {}", fom, tom, total)
        return total
    }

```

Reply:
Jeg tror du kan fjerne `if (erSammeMånedOgÅr) {` fra `beregnSisteMånedProsent` uten å legge på toppnivå og få samme resultat.

Reply:
Bra forslag, men tester feiler

Reply:
Ahh.. Ser greia.. beregnFørsteMånedProsent har en annen logikk rundt "erSammeMånedOgÅr" enn jeg hadde foreslått.